src/main/java/ontos/task/TaskList.java

```diff
@@ -0,0 +1,145 @@
+     */
+    public void addTask(Task task) {
+        this.taskList.add(task);
+    }
+
+    /**
+     * Marks the task at the specified index as completed.
+     *
+     * @param index The index of the task to be marked as completed (1-based index).
+     * @throws IndexOutOfBoundsException If the index is out of range (index < 1 or index > size()).
+     */
+    public void completeTaskAt(int index) throws IndexOutOfBoundsException {
+        this.taskList.get(index - 1).completeTask();
+    }
+
+    /**
+     * Marks the task at the specified index as incomplete.
+     *
+     * @param index The index of the task to be marked as incomplete (1-based index).
+     * @throws IndexOutOfBoundsException If the index is out of range (index < 1 or index > size()).
+     */
+    public void uncompleteTaskAt(int index) throws IndexOutOfBoundsException {
+        this.taskList.get(index - 1).uncompleteTask();
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return The number of tasks in the list.
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Removes the task at the specified index from the list and returns it.
+     *
+     * @param index The index of the task to be removed (1-based index).
+     * @return The removed task.
+     * @throws IndexOutOfBoundsException If the index is out of range (index < 1 or index > size()).
+     */
+    public Task removeTaskAt(int index) throws IndexOutOfBoundsException {
+        Task removedTask = taskList.get(index - 1);
+        taskList.remove(index - 1);
+        return removedTask;
+    }
+
+    /**
+     * Returns the task at the specified index.
+     *
+     * @param index The index of the task to be retrieved (1-based index).
+     * @return The task at the specified index.
+     * @throws IndexOutOfBoundsException If the index is out of range (index < 1 or index > size()).
+     */
+    public Task getTaskAt(int index) {
+        return taskList.get(index - 1);
+    }
+
+    /**
+     * Returns the string representation of the task at the specified index.
+     *
+     * @param index The index of the task to be converted to a string (1-based index).
+     * @return The string representation of the task.
+     * @throws IndexOutOfBoundsException If the index is out of range (index < 1 or index > size()).
+     */
+    public String getTaskString(int index) {
+        return taskList.get(index - 1).toString();
+    }
+
+    /**
+     * Returns a string representation of the task list, with each task on a new line.
+     *
+     * @return The string representation of the task list.
+     */
+    @Override
+    public String toString() {
+        StringBuilder output = new StringBuilder();
+        for (int i = 0; i < this.getSize(); i++) {
+            int index = i + 1;
+            output.append(" ").append(index).append(". ").append(taskList.get(i).toString());
+            if (index < this.getSize()) {
```

Comment:
I like how you use an if-else statement to avoid adding an empty line at the end of the file.